src/main/kotlin/uk/gov/justice/digital/hmpps/hmppschallengesupportinterventionplanapi/controller/CsipRecordsController.kt

```diff
@@ -37,6 +38,39 @@ import java.util.UUID
 @RequestMapping(produces = [MediaType.APPLICATION_JSON_VALUE])
 @Tag(name = "1. CSIP Record Controller", description = "Endpoints for CSIP Record operations")
 class CsipRecordsController(val csipRecordService: CsipRecordService) {
+
+  @Operation(
+    summary = "Retrieve the current CSIP record for a prisoner.",
+    description = "Returns a summary of the current CSIP record and counts of opened csip and referrals for the prisoner",
+  )
+  @ApiResponses(
+    value = [
+      ApiResponse(
+        responseCode = "200",
+        description = "CSIP records found",
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = "Bad Request",
+        content = [Content(schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorised, requires a valid Oauth2 token",
+        content = [Content(schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Forbidden, requires an appropriate role",
+        content = [Content(schema = Schema(implementation = ErrorResponse::class))],
+      ),
+    ],
+  )
+  @GetMapping("/prisoners/{prisonNumber}/csip-records/current")
+  @PreAuthorize("hasAnyRole('$ROLE_CSIP_UI')")
```

Comment:
I missed this in the requirements. We'll need to add a `ROLE_CHALLENGE_SUPPORT_INTERVENTION_PLAN__RO` role and grant it to `hmpps-prisoner-profile-system` so the Prisoner Profile can use this endpoint. Don't want to give it the UI's role